MapboxCoreNavigation/LegacyRouteController.swift

```diff
@@ -478,8 +478,11 @@ open class LegacyRouteController: NSObject, Router, InternalRouter, CLLocationMa
         for voiceInstruction in spokenInstructions {
             if userSnapToStepDistanceFromManeuver <= voiceInstruction.distanceAlongStep || firstInstructionOnFirstStep {
 
+                let currentSpokenInstruction = routeProgress.currentLegProgress.currentStepProgress.currentSpokenInstruction!
```

Comment:
The current spoken instruction is already stored in the (somewhat misnamed) local variable `voiceInstruction`. Use that variable to avoid force-unwrapping `currentSpokenInstruction`, which has been the source of crashes in the past.

Reply:
Indeed, I've applied the same recommandation to the updateVisualInstructionProgress method.